### ISSUE TITLE
Previous tab focusing after tab closing or tab unloading

### ISSUE THREAD

**Issue by N3C2L** [madprops/grasshopper#127]
It would be nice if there could be two small additions to the handling of tabs. The current problem is that when you are inside an active tab, and this tab will then be closed or unloaded inside the grasshopper interface, the next tab below is always activated. However, this becomes an issue when the next tab is unloaded. This tab then gets loaded even though that's not always desired. Therefore, it would be great if there were an option to determine what should happen after the active tab is closed or unloaded. In my opinion, the best solution in both cases would be to focus on the most recently selected active loaded tab.

**Comment by madprops** [comment]
Grasshopper controls what tab gets activated when the active one gets unloaded. And it goes down skipping unloaded tabs until it finds a normal one. So not sure what you mean there. About closing tabs, that's something the browser does and I can't control it afaik but I'll keep it in mind.

**Comment by N3C2L** [comment]
Thanks for looking into it. What I meant was, when you are currently in the active tab, after unloading or closing this active tab, the previous active loaded tab should be focused.

**Comment by madprops** [comment]
Let me know if it works better now.

Added `Smart Tab Switch`, enabled by default. For now it only controls closing tabs.

Unloading tabs always gets the unload switch.

**Comment by N3C2L** [comment]
Thank you, this is really useful! The Smart Tab Switch is working fine!
I know it sounds strange, but neither in sidebery nor in the tree style addon is there an explicit option to set what should happen after unloading a tab. In the tree style tabs addon, there isn't even a command for unloading. But in sidebery, when you unload a tab through its interface, it automatically switches to the previously activated tab. However, I don't understand why sidebery doesn't mention this anywhere in its options. Anyway, it would be really great if you could look into what sidebery does differently when a tab is unloaded. So, why does it switch to the previous active tab after unloading, because that would be a really nice option to have for grasshopper.

**Comment by madprops** [comment]
GH does the same thing when unloading the tab except the direction is different, it doesn't go to previously used tab, but check downwards for a normal loaded tab and switch to it. If not found, it searches upwards. If not found, it opens a new tab. This is not working for you? I wonder if some setting is messing it up.

**Comment by N3C2L** [comment]
No, it's working. No problem here. I was just thinking it would helping me in this case: For instance if I load a tab in another zone, just one tab, and then I want to unload this, it should me bring back to the previous zone. That would be useful here so I don't need to navigate back to the zone/tab I was before.

**Comment by madprops** [comment]
Ok i'll look into traversing back in tab history to select the next tab. I guess you'd want that behavior on unload and close tabs.

**Comment by N3C2L** [comment]
Great! Thanks!

**Comment by madprops** [comment]
Should be working now on `v3206`

**Comment by N3C2L** [comment]
Can confirm! 
Many thanks, you are a great help!
I think I can close this.

**Comment by madprops** [comment]
There was an issue since it was still considering unloaded tabs for the prev tab. It now filters out unloaded.

**Comment by N3C2L** [comment]
Perfect, thanks!

**Comment by N3C2L** [comment]
Please excuse me if I may suggest an improvement in this thread. You have already kindly made it so that when an an active tab is unloaded, it switches to the previous active tab. But would it be possible to do the same for an active tab when it is closed? That would be fine. You could perhaps check the code in sidebery, as it works there:
<img width="474" alt="sidebery" src="https://github.com/user-attachments/assets/45fa32f7-a0ed-49d6-9925-d89a441c90da">

**Comment by madprops** [comment]
It already does this.

**Comment by N3C2L** [comment]
My bad, sorry 😢 
I have in Settings>Triggers>Middle Click Press no cmd assigned, and could close the tab with the middle mouse button. Could you reproduce that? Try to unassign the "Middle Click Press" to no cmd and try to close a tab with the middle mouse button. It should close the tab. But if I closed it that way without the "close" cmd, then it wasn't activate the previous tab. So your implementation of the previous tab switch is working when I have assigned the "close" cmd on the "Middle Click Press".

**Comment by madprops** [comment]
`Middle Click Press` in triggers refers to a long press of middle click.

What you're looking for is `Middle Click Tabs` in auxclick.

**Comment by N3C2L** [comment]
Ok this is weird. Because I already had assign this Middle Click Tabs in auxclick to the close cmd.
I had to assign the close cmd also to middle click press in order to make your previous tab switch work. Maybe there is something with my mouse and delay settings.

**Comment by madprops** [comment]
Make sure `Smart Tab Switch` is enabled in More.

**Comment by N3C2L** [comment]
Thanks, I reassigned the close cmd in auxclick and reenabled Smart Tab Switch, and now it works. Yes it's a middle click button. But I don't have to to assign a key in middle click press anymore. But I was already thinking that it was related to this software I use
https://www.highrez.co.uk/downloads/XMouseButtonControl.htm
Because I have assigned my third mouse button to mouse button 4 (thumb key) with this software.
But know everything is working, sorry for the false alarm 😆
